Source/ACE/Entity/Door.cs

```diff
@@ -242,35 +246,42 @@ public override void OnUse(ObjectGuid playerId)
 
         private void Open(ObjectGuid opener = new ObjectGuid())
         {
-            CurrentLandblock.EnqueueBroadcastMotion(this, motionOpen);
-            CurrentMotionState = motionStateOpen;
-            PhysicsState |= PhysicsState.Ethereal;
-            Ethereal = true;
-            IsOpen = true;
-            if (opener.Full > 0)
-                UseTimestamp++;
+            ActionChain chain = new ActionChain();
```

Comment:
This is closer -- but your still checking IsLocked and IsOpen in the Player's scope, not the doors (in OnUse).